more_itertools/more.py

```diff
@@ -3447,3 +3448,40 @@ def windowed_complete(iterable, n):
         middle = seq[i : i + n]
         end = seq[i + n :]
         yield beginning, middle, end
+
+
+def all_unique(iterable, key=None):
+    """
+    Returns ``True`` if all the elements of *iterable* are unique (no two
+    elements are equal).
+
+    If a *key* function is specified, it will be used to preprocess the
+    elements of the iterable before comparing.
+
+        >>> all_unique('ABCB')
+        False
+        >>> all_unique('ABCb')
+        True
+        >>> all_unique('ABCb', str.lower)
+        False
+
+    Sequences with a mix of hashable and unhashable items can be used, but the
+    function will be slower for unhashable items. This is similar to
+    :func:`unique_everseen`, see details there.
+    """
+    seenset = set()
```

Comment:
To make the reasoning more clear, I'd favor using the names `seen_hashable` (`seenset`) and `seen_unhashable` (`seenlist`) instead.

Reply:
again, copied the implementation. do you think the improvement in clarity outweighs the benefit of the implementation being as similar as possible to `unique_everseen`?